App creation time out with the latest ignite cli

### Describe the bug

Hi Team, first time I am using the ignite cli and I used this command `npx ignite-cli@latest new SocialSyncer --no-timeout --yes`. Initially I used without the `timeout` option and got an error `Error: App creation timed out.` and then I used the `--no-timeout` flag and getting the same error. I am using all default options.

```
npx ignite-cli@latest new SocialSyncer --no-timeout --yes





                                               .;;,.    .clll.
                                              .clll,    ,llll.           ...
                                               .,;,.    ;lllc'.,.   .';cllllc,
     .',,.                           ...      .'',,. .:lllllllll. .;lllc:;clll,
    .llll,                    .,...;cllll:.   ,llll, .llllllc,,'  clll:'.,cllc.
    ,llll'       ......;::.  'llcclllllllll'  ,llll,   .llll.    ,lllllllll:'
    ,llll.    .;clllllclll.  'llllc,..,llll,  :llll.   'llll.    ,lllc,''..
    ;llll.   :llllc:cllllc.  ,llll.   .llll, .lllll.   ,lllc...  ,lll:...';:.
   .llllc.  ;lll;.   :lll,   ;lll;    'llll, .llllc.   ,llllcll' .:lllllllll,
   .llll,  'lll:    .clll,   :lll,    ,llll. 'llll,    'lllllcc'   ';::::;'.
   .llll,  ,lll:    .llll'  .llll,    ;llll. ,llll,     .....             .',..
   ,llll.  'llll:,,:cllll.  .lllc.    .''..  ... ....',;         ..',,:::;'..
   ,llll.   'clllllllllll.     .....        ..,::cllllc.   .',:clllc;,'.
   :llll.     ',;,'.;lllc...,;:clll;  ..',:cllllllllll:,;clllllc;,'.
   llll:        ...'cllllllllllllll::cllllllc;,',cllllllllc;'..
   llll,   ..,clllllllll::,,.,lllllllc:,'...    :lllllc,..
   c:;,.  .cllllc:;:llll.    ,ll:;,..          ;ll:,..
          :lll,.  .clll,     ..               .,..
          :lll:,,cllll,
          .cllllllllc'
           .'cllll:.



    █ Creating SocialSyncer using Ignite 11.1.2
    █ Powered by  ∞ Infinite Red  (https://infinite.red)
    █ Package Manager: yarn
    █ Bundle identifier: com.socialsyncer
    █ Path: /Users/athondapu/devhome/workspaces/SocialSyncer
    ────────────────────────────────────────────────────

    🖨  3D-printing a new React Native app
    🎨 Getting those last few details perfect
    🧶 Installing yarn dependencies (wow these are heavy)
    ⚙️  Configuring app.json
    ⠙  Generating native template via Expo Prebuild
   Error: App creation timed out.
   Run again with --debug to see what's going on.
```

I tried multiple times but still getting the same error. Also, it's failing for both package managers `yarn` and `npm`.

I am using the latest system (`Apple M3 Pro`) and have a `100Mbps` internet speed but still getting the timeout issue.

```
Node Version: v22.19.0
NPM Version: 10.9.3
```

### Ignite version

11.1.2

### Additional info

```
npx ignite-cli doctor
System
  platform           darwin
  arch               arm64
  cpu                12 cores     Apple M3 Pro
  directory          workspaces   /Users/athondapu/devhome/workspaces

JavaScript (and globally-installed packages)
  node               22.19.0      /Users/athondapu/.nvm/versions/node/v22.19.0/bin/node
  npm                10.9.3       /Users/athondapu/.nvm/versions/node/v22.19.0/bin/npm
    corepack         0.34.0
    expo-cli         6.3.12
    npm              10.9.3
  yarn               1.22.22      /Users/athondapu/.asdf/shims/yarn
  pnpm               -            not installed
  bun                -            not installed

Ignite
  ignite-cli         11.1.2       /Users/athondapu/.npm/_npx/e31027f3785124a8/node_modules/.bin/ignite
  ignite src         build        /Users/athondapu/.npm/_npx/e31027f3785124a8/node_modules/ignite-cli/build

Android
  java               17.0.16      /opt/homebrew/opt/openjdk@17/bin/java
  android home       -            /Users/athondapu/Library/Android/sdk

iOS
  xcode              16.4
  cocoapods          1.16.2       /opt/homebrew/bin/pod

Tools
  git                git version 2.51.0   /opt/homebrew/bin/git
```

HI Team, now the app was created but I still see some errors while setup the project. Here are some samples

```
🖨  3D-printing a new React Native app
    🎨 Getting those last few details perfect
    🧶 Installing yarn dependencies (wow these are heavy)
    ⚙️  Configuring app.json
    🛠️  Generating native template via Expo Prebuild
    🛠️  Removing fancy demo markup
    ❌ Cleaning up

The following error occurred:

    yarn run v1.22.22
$ eslint . --fix

/Users/athondapu/devhome/workspaces/SocialSyncer/.eslintrc.js
  1:1  error  Definition for rule 'reactotron/no-tron-in-production' was not found  reactotron/no-tron-in-production

/Users/athondapu/devhome/workspaces/SocialSyncer/app.config.ts
  1:1  error  Definition for rule 'reactotron/no-tron-in-production' was not found  reactotron/no-tron-in-production

/Users/athondapu/devhome/workspaces/SocialSyncer/app/app.tsx
  1:1  error  Definition for rule 'reactotron/no-tron-in-production' was not found  reactotron/no-tron-in-production

/Users/athondapu/devhome/workspaces/SocialSyncer/app/components/AutoImage.tsx
  1:1  error  Definition for rule 'reactotron/no-tron-in-production' was not found  reactotron/no-tron-in-production

/Users/athondapu/devhome/workspaces/SocialSyncer/app/components/Button.tsx
  1:1  error  Definition for rule 'reactotron/no-tron-in-production' was not found  reactotron/no-tron-in-production

/Users/athondapu/devhome/workspaces/SocialSyncer/app/components/Card.tsx
  1:1  error  Definition for rule 'reactotron/no-tron-in-production' was not found  reactotron/no-tron-in-production
```

Can anyone please help me on this.?

--- a/src/commands/new.ts
+++ b/src/commands/new.ts
@@ -486,7 +486,7 @@ module.exports = {
     const perfStart = new Date().getTime()
 
     // add a timeout to make sure we don't hang on any errors
-    const timeoutHandle = noTimeout ? undefined : setTimeout(timeoutExit, MAX_APP_CREATION_TIME)
+    const timeoutHandle = !noTimeout && setTimeout(timeoutExit, MAX_APP_CREATION_TIME)
 
     // #region Print Welcome
     // welcome everybody!

